src/agent.cpp

```diff
@@ -882,17 +906,21 @@ Agent::doPassTo(int receiver)
 bool
 Agent::doDribble()
 {
+  bool success = false;
   Strategy::instance().update( world() );
   M_field_evaluator = createFieldEvaluator();
   CompositeActionGenerator * g = new CompositeActionGenerator();
   g->addGenerator(new ActGen_MaxActionChainLengthFilter(new ActGen_ShortDribble(), 1));
   M_action_generator = ActionGenerator::ConstPtr(g);
   ActionChainHolder::instance().setFieldEvaluator( M_field_evaluator );
   ActionChainHolder::instance().setActionGenerator( M_action_generator );
-  doPreprocess();
+  success = doPreprocess();
   ActionChainHolder::instance().update( world() );
-  Bhv_ChainAction(ActionChainHolder::instance().graph()).execute(this);
-  return true;
+  if (Bhv_ChainAction(ActionChainHolder::instance().graph()).execute(this)) {
```

Comment:
if (Bhv_ChainAction(ActionChainHolder::instance().graph()).execute(this) || successful_preprocess) {
return true;
}
return false;